shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/repositories/PredictionsRepository.kt

```diff
@@ -46,6 +54,29 @@ class PredictionsRepository(private val socket: PhoenixSocket) :
             ?.receive(PhoenixPushStatus.Error) { onReceive(Outcome(null, SocketError.Connection)) }
     }
 
+    override fun connectV2(
+        stopIds: List<String>,
+        onJoin: (Outcome<PredictionsByStopJoinResponse?, SocketError>) -> Unit,
+        onMessage: (Outcome<PredictionsByStopMessageResponse?, SocketError>) -> Unit,
+    ) {
+        disconnect()
+        channel = socket.getChannel(PredictionsForStopsChannel.topicV2(stopIds), mapOf())
```

Comment:
I'm concerned about a race condition here if connectV2 is called in quick succession within a single PredictionsRepository. Are we disconnecting from the right channel?